nexusutils/dataconverter/helpers.py

```diff
@@ -466,7 +466,22 @@ def check_for_valid_atom_types(atoms: Union[str, list]):
                              f"check for correct element name")
 
 
-def extract_atom_types(formula: str, mode: str = 'hill') -> Union[Set, List]:
+def convert_to_hill(atoms_typ):
+    """Convert list of atom into to hill."""
+    if not isinstance(atoms_typ, list) and isinstance(atoms_typ, set):
```

Comment:
there should be a check which catches the case when atoms_typ is empty or neither list nor set

Reply:
what if atoms_typ has D (deuterium)?